06-building-a-program-start-to-finish/writing-tests/output/invertedSpec.js

```diff
@@ -0,0 +1,152 @@
+        text: 'Alice falls into a rabbit hole.' }),
+        Object({ title: 'The Lord of the Rings',
+          text: 'An unusual alliance of man.' }),
+        Object({ title: 'The Lord of the Rings:',
+          text: 'An unusual alliance of man.' })
+      ] };
+    expect(InvertedIndex.validateFile(smallBook)).toEqual(successMsg);
+  });
+
+  it('should return false for files without "title" and "text" properties',
+     () => {
+       const msg = false;
+       expect(InvertedIndex.validateFile(invalidBook)).toEqual(msg);
+     });
+
+  it('should return false if file is not an array of JSON object',
+     () => {
+       expect(InvertedIndex.validateFile(invalidBook)).toBeFalsy();
+     });
+
+  it('should return false if file contains an empty array',
+     () => {
+       const msg = { status: false };
+       expect(InvertedIndex.validateFile(malformedJSON)).toEqual(msg);
+     });
+
+  describe('Create Index', () => {
+    it('should return mapped indices to words in a JSON file', () => {
+      const expectedResult =
+        { 'smallValidBook.json':
+        { a: [0],
+          alice: [0],
+          falls: [0],
+          hole: [0],
+          in: [0],
+          into: [0],
+          rabbit: [0],
+          wonderland: [0],
+          alliance: [1, 2],
+          an: [1, 2],
+          lord: [1, 2],
+          man: [1, 2],
+          of: [1, 2],
+          rings: [1, 2],
+          the: [1, 2],
+          unusual: [1, 2] } };
+      expect(invertedIndex.createIndex(smallBook, 'smallValidBook.json')).toEqual(expectedResult);
+    });
+  });
+
+  describe('Tokenize words', () => {
+    it('should strip out special characters from excerpt in documents', () => {
+      let excerpt = 'Alice loves her ima&&gination';
+      const expectedTokens = 'Alice loves her imagination';
+      excerpt = InvertedIndex.tokenizeWords(excerpt);
+      expect(expectedTokens).toEqual(excerpt);
+    });
+  });
+
+
+  describe('Get Index', () => {
+    it('should verify that index has been created', () => {
+      expect(Object.keys(invertedIndex.getIndex('smallValidBook.json')).length)
+        .toBeGreaterThan(0);
+    });
+  });
+
+  describe('Search index', () => {
+    it('should return true if search term is a string', () => {
+      const term = 'Wonderland of rings';
+      expect(Object.keys(invertedIndex.searchIndex(term, 'smallValidBook.json')))
+      .toBeTruthy();
+    });
+  });
+
+  it('should search through single files that are indexed', () => {
+    const requiredOutput = {
+      'smallValidBook.json':
+      {
+        alice: [0],
+        and: [],
+        her: [],
+        imagination: [],
+        unusual: [1, 2]
+      }
+    };
+    let searchTerm = {};
+    searchTerm = invertedIndex.searchIndex('Alice, and her unusual imagination',
+      'smallValidBook.json');
+    expect(Object.keys(searchTerm)).toEqual(Object.keys(requiredOutput));
+    expect(searchTerm).toEqual(requiredOutput);
+  });
+});
```

Comment:
I have now included test suites for all but one of my methods (i.e. 'readFile' method).
For the readFile method, I used HTML5 fileReader API which currently cannot be tested because it requires an actual file upload.